types/styletron-react/index.d.ts

```diff
@@ -0,0 +1,271 @@
+        style: (props: P) => StyleObject
+    ): StyletronComponent<
+        Pick<
+            React.ComponentProps<C>,
+            Exclude<keyof React.ComponentProps<C>, { className: string }>
+        > &
+            P
+    >;
+    <C extends keyof JSX.IntrinsicElements | React.ComponentType<any>>(
+        component: C,
+        style: StyleObject
+    ): StyletronComponent<
+        Pick<
+            React.ComponentProps<C>,
+            Exclude<keyof React.ComponentProps<C>, { className: string }>
+        >
+    >;
+}
+
+export interface WithStyleFn {
+    <C extends StyletronComponent<any>, P extends object>(
+        component: C,
+        style: (props: P) => StyleObject
+    ): StyletronComponent<React.ComponentProps<C> & P>;
+    <C extends StyletronComponent<any>>(
+        component: C,
+        style: StyleObject
+    ): StyletronComponent<React.ComponentProps<C>>;
+}
+
+export interface WithTransformFn {
+    <C extends StyletronComponent<any>, P extends object>(
+        component: C,
+        style: (style: StyleObject, props: P) => StyleObject
+    ): StyletronComponent<React.ComponentProps<C> & P>;
+}
+
+export interface WithWrapperFn {
+    <C extends StyletronComponent<any>, P extends object>(
+        component: C,
+        wrapper: (component: C) => React.ComponentType<P>
+    ): StyletronComponent<React.ComponentProps<C> & P>;
+}
+
+// ^^^^^^^^^^^^^^^^^^^^^^^^^
+// End of style-types
+
+export class BrowserDebugEngine {
+    debug(stack: { stackIndex: StackIndex; stackInfo: StackInfo }): string;
+}
+
+export class NoopDebugEngine {
+    debug(): void;
+}
+
+export type DebugEngine = BrowserDebugEngine | NoopDebugEngine;
+
+export interface DevProviderProps {
+    children: React.ReactNode;
+    value: StandardEngine;
+    debugAfterHydration?: boolean;
+    /** DebugEngineContext */
+    debug?: DebugEngine;
+}
+
+export class DevProvider extends React.Component<
+    DevProviderProps,
+    { hydrating: boolean }
+> {}
+
+export const Provider: typeof DevProvider | React.Provider<StandardEngine>;
+
+export function DevConsumer(props: {
+    children: (
+        styletronEngine: StandardEngine,
+        debugEngine: DebugEngine,
+        hydrating: boolean
+    ) => React.ReactNode;
+}): JSX.Element;
+
+/**
+ * @param style the StyleObject
+ * @returns string to be used in className prop of JSX.Element
+ */
+export type CSSFn = (style: StyleObject) => string;
+
+export function useStyletron(): Readonly<[CSSFn]>;
+
+export interface CreateStyledOptions {
+    getInitialStyle: StyletronGetInitialStyle;
+    driver: StyletronDriver;
+    wrapper: StyletronWrapper;
+}
+
+export function createStyled(options: CreateStyledOptions): StyledFn;
+
+export const styled: ReturnType<typeof createStyled>;
+
+export function createStyledElementComponent(
+    styletron: Styletron
+): StyletronComponent<any>;
+
+// export const withTransform: WithTransformFn;
+export function withTransform<C extends StyletronComponent<any>, P>(
```

Comment:
Is there a way to directly use the `WithTransformFn` interface when describing this function? Instead of basically duplicating the typing?

Reply:
probably `export const withTransform: WithTransformFn`